File: hashtable/136.c
/*
 * tricky的方法, 两个相同的数座xor(^操作)是0, 而0^任何数都等于任何数
 */

int singleNumber(int* nums, int numsSize) {
    int result = nums[0];
    for(int i = 1; i < numsSize; i++)
    {
        result ^= nums[i];
    }
    return result;
}

File: bit_manipulation/136.py
#math solution
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        return sum(set(nums))*2 - sum(nums)


#xor solution, 注意xor的性质 1 ^ 1 = 0, 1 ^ 0 = 1
import functools
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        return functools.reduce(lambda a, b: a ^ b, nums)
